main.tf

```diff
@@ -1,13 +1,254 @@
+  iam_role_enabled       = local.enabled && var.create_iam_role
+  account_id             = data.aws_caller_identity.current.account_id
+  region                 = data.aws_region.current.name
+}
+
+module "s3_label" {
+  source  = "cloudposse/label/null"
+  version = "0.25.0"
+  enabled = local.s3_bucket_enabled
+
+  attributes = ["s3"]
+
+  context = module.this.context
+}
+
+module "sg_label" {
+  source  = "cloudposse/label/null"
+  version = "0.25.0"
+  enabled = local.security_group_enabled
+
+  attributes = ["sg"]
+
+  context = module.this.context
+}
+
+module "iam_label" {
+  source  = "cloudposse/label/null"
+  version = "0.25.0"
+  enabled = local.iam_role_enabled
+
+  attributes = ["iam"]
+
+  context = module.this.context
+}
+
+data "aws_iam_policy_document" "iam_policy_document" {
+  statement {
+    sid       = ""
+    actions   = ["airflow:PublishMetrics"]
+    effect    = "Allow"
+    resources = ["arn:aws:airflow:${local.region}:${local.account_id}:environment/${module.this.id}"]
+  }
+
+  statement {
+    sid     = ""
+    actions = ["s3:ListAllMyBuckets"]
+    effect  = "Allow"
+    resources = [
+      var.create_s3_bucket ? module.mwaa_s3_bucket.bucket_arn : var.source_bucket_arn,
+      "${var.create_s3_bucket ? module.mwaa_s3_bucket.bucket_arn : var.source_bucket_arn}/*"
+    ]
+  }
+
+  statement {
+    sid = ""
+    actions = [
+      "s3:GetObject*",
+      "s3:GetBucket*",
+      "s3:List*"
+    ]
+    effect = "Allow"
+    resources = [
+      var.create_s3_bucket ? module.mwaa_s3_bucket.bucket_arn : var.source_bucket_arn,
+      "${var.create_s3_bucket ? module.mwaa_s3_bucket.bucket_arn : var.source_bucket_arn}/*"
+    ]
+  }
+
+  statement {
+    sid       = ""
+    actions   = ["logs:DescribeLogGroups"]
+    effect    = "Allow"
+    resources = ["*"]
+  }
+
+  statement {
+    sid = ""
+    actions = [
+      "logs:CreateLogStream",
+      "logs:CreateLogGroup",
+      "logs:PutLogEvents",
+      "logs:GetLogEvents",
+      "logs:GetLogRecord",
+      "logs:GetLogGroupFields",
+      "logs:GetQueryResults",
+      "logs:DescribeLogGroups"
+    ]
+    effect    = "Allow"
+    resources = ["arn:aws:logs:${local.region}:${local.account_id}:log-group:airflow-${module.this.id}*"]
+  }
+
+  statement {
+    sid       = ""
+    actions   = ["cloudwatch:PutMetricData"]
+    effect    = "Allow"
+    resources = ["*"]
+  }
+
+  statement {
+    sid = ""
+    actions = [
+      "sqs:ChangeMessageVisibility",
+      "sqs:DeleteMessage",
+      "sqs:GetQueueAttributes",
+      "sqs:GetQueueUrl",
+      "sqs:ReceiveMessage",
+      "sqs:SendMessage"
+    ]
+    effect    = "Allow"
+    resources = ["arn:aws:sqs:${local.region}:*:airflow-celery-*"]
+  }
+
+  statement {
+    sid = ""
+    actions = [
+      "kms:Decrypt",
+      "kms:DescribeKey",
+      "kms:GenerateDataKey*",
+      "kms:Encrypt"
+    ]
+    effect        = "Allow"
+    not_resources = ["arn:aws:kms:*:${local.account_id}:key/*"]
```

Comment:
please add some comments on what this statement does (not easy to understand by just looking at it since it has/uses `not_resources` and `sqs`

Reply:
Removed to resources instead of not_resources per https://docs.aws.amazon.com/mwaa/latest/userguide/mwaa-create-role.html